src/linker/Linker.Steps/BlacklistStep.cs

```diff
@@ -55,7 +55,7 @@ protected override void Process ()
 					steps_to_add.Push (GetResolveStep (name));
 				} catch (XmlException ex) {
 					/* This could happen if some broken XML file is included. */
-					Context.LogMessage ($"Error processing {name}: {ex}");
+					throw new LinkerErrorException (MessageContainer.CreateErrorMessage ($"Error processing {name}: {ex}", 1003));
```

Comment:
Generally speaking I think it's good to recover from errors if we have a reasonable way to do so, as long as an error still fails the build. That way you can fix more problems in one iteration.

Reply:
Yes, this is a bad pattern. We want to produce as many errors possible during the linker run not abort on the first one.

Reply:
I have to apologize to Mateo - this was my suggestion. I thought that we use warnings in places where linker can safely recover from it. If there are places where we want the build to fail, but linker can safely recover, then I agree we should not throw there.